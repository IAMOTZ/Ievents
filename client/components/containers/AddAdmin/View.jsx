import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import Footer from '../../common/Footer';
import { AuthTopNavigation } from '../../common/TopNavigation';
import { LoadingIcon } from '../../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../../common/Alert';

const View = props => (
  <div id="add-admin-container">
    {/* Top navigation on small screen */}
    <SideNavigation
      name={props.userName}
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        {/*  Side navigation on large screen */}
        <AuthTopNavigation
          name={props.userName}
          title="Add an admin"
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-md-2 mt-lg-0">
          <Header text="Add an admin" />
          <div className="page-content">
            <div className="d-flex flex-column align-items-center mt-5">
              <LoadingIcon start={props.addingAdminStarted} size={3} />
              <div className="card align-items-center text-center w-75">
                <div className="card-body">
                  <BigAlert message={props.addingAdminError} />
                  {
                    props.addingAdminResolved ?
                      <BigAlert
                        message={`${props.newAdmin} is now an admin`}
                        type="success"
                      /> : null
                  }
                  <div className="input-group px-3">
                    <span className="input-group-addon">
                      <i className="fa fa-envelope" />
                    </span>
                    <input
                      id="input-email"
                      type="text"
                      className="form-control"
                      placeholder="The user's email"
                      name="email"
                      onChange={props.getInput}
                    />
                  </div>
                  <SmallAlert message={props.inputErrors.emailError} />
                  <ul className="text-left mt-2">
                    <li>The user must have signed up already</li>
                    <li>If successful, the user would have access to:
                      <ol type="number">
                        <li>Creating a Center</li>
                        <li>Editing a Center</li>
                        <li>Viewing transactions</li>
                        <li>Making decisions on transactions</li>
                      </ol>
                    </li>
                  </ul>
                  <button
                    id="add-btn"
                    className="btn ie-blue-button"
                    disabled={props.addingAdminStarted}
                    onClick={props.add}
                  >Add Admin
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span className="d-block d-sm-none mt-5">
      <Footer />
    </span>
  </div>
);


View.defaultProps = {
  newAdmin: '',
  addingAdminError: '',
  inputErrors: {},
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  inputErrors: PropTypes.object,
  addingAdminStarted: PropTypes.bool.isRequired,
  addingAdminError: PropTypes.string,
  addingAdminResolved: PropTypes.bool.isRequired,
  newAdmin: PropTypes.string,
  getInput: PropTypes.func.isRequired,
  add: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default View;
