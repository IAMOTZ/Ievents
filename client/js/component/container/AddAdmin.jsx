import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import { addAdmin, clearStatus } from '../../actions/authAction';

import UserSideNav from '../common/SideNavigation.jsx';
import Header from '../common/Header.jsx';
import { UserTopNav } from '../common/TopNavigation.jsx';

@connect((store) => {
  return {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    status: {
      success: store.user.status.adminAdded,
      error: store.user.status.addingAdminError.message
    }
  }
})

export default class AddAdmin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      inputError: null,
    }
  }

  refresh() {
    this.props.dispatch(clearStatus('ADD_ADMIN'));
  }

  getInput = (e) => {
    if (this.props.status.success) {
      this.refresh();
    }
    const state = this.state;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  add = (e) => {
    if (this.state.email) {
      this.setState({ inputError: null });
      this.props.dispatch(addAdmin(this.state.email, this.props.user.token));
    } else {
      this.setState({ inputError: 'Email is required' });
    }
  }

  render() {
    if (!this.props.authenticated) {
      return (<Redirect to="/users/login" />);
    } else {
      return (
        <div id="add-admin-container">
          {/* Top navigation on small screen */}
          <UserTopNav name={this.props.user.name} title='Add an admin' />
          <div className="container-fluid">
            <div className="row">
              {/*  Side navigation on large screen */}
              <UserSideNav userName={this.props.user.name} />
              {/* Main content */}
              <div className="col-lg-10 offset-md-2 mt-lg-0" id="main-content">
                {/* Content Header(navigation) on large screen */}
                <Header text='Add an admin' />
                <div className="card text-center align-items-center mt-5 w-75 mx-auto">
                  <div className="card-body">
                    <Alert
                      inputError={this.state.inputError}
                      addingError={this.props.status.error}
                      success={this.props.status.success}
                      newAmdin={this.state.email} />
                    <div className="input-group px-3">
                      <span className="input-group-addon">
                        <i className="fa fa-user"></i>
                      </span>
                      <input
                        type="text"
                        className="form-control"
                        placeholder="The user's email"
                        name="email"
                        onChange={this.getInput} />
                    </div>
                    <ul className="text-left mt-2 text-muted">
                      <li>The user must have signed up already</li>
                      <li>If successful, the user would have access to:
                        <ol type="number">
                          <li>Creating a Center</li>
                          <li>Editing a Center</li>
                          <li>Viewing transactions</li>
                          <li>Making decisions on transactions</li>
                        </ol>
                      </li>
                    </ul>
                    <button
                      className="btn btn-primary"
                      onClick={this.add}>Add Admin</button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <footer className="d-block d-sm-none mt-5">
            <div className="container text-white text-center py-5">
              <h1>Ievents</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </footer>
        </div>
      );
    }
  }
}

const Alert = (props) => {
  if (props.addingError || props.inputError) {
    return (
      <p className="text-danger">{props.inputError || props.addingError}</p>
    );
  } else if (props.success) {
    return (
      <span>
        <p className="text-success">Admin Added!</p>
        <span>{props.newAmdin} is now and admin</span>
      </span>
    );
  } else {
    return null;
  }
}
