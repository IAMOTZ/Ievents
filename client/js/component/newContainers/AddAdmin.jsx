import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { validateAddAdminInputs } from '../../helpers/inputValidators';
import { addAdmin } from '../../actions/authAction';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import UserSideNav from '../common/SideNavigation';
import Header from '../common/Header';
import { UserTopNav } from '../common/TopNavigation';
import { LoadingIcon } from '../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../common/Alert';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    user,
    isAdmin: user.role === 'admin',
    isSuperAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isUserAuthenticaed: store.authReducer.loggingUserResolved,
    addingAdminStarted: store.addAdminReducer.addingAdminStarted,
    addingAdminResolved: store.addAdminReducer.addingAdminResolved,
    addingAdminError: store.addAdminReducer.addingAdminError,
  };
})
class AddAdmin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      inputErrors: {
        emailError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_ADMIN));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    if (this.props.addingAdminResolved || this.props.addingAdminError) {
      this.refresh();
    }
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      emailError: null,
    };
    this.setState(state);
  }

  /**
   * Refresh the page by stopping any asynchronous process that might be going on.
   */
  refresh() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_ADMIN));
  }

  /**
   * Dispatches the action to add the admin.
   */
  add = () => {
    const { email } = this.state;
    const inputErrors = validateAddAdminInputs({ email });
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(addAdmin(this.state.email, this.props.user.token));
    }
  }

  render() {
    if (!this.props.isUserAuthenticaed) {
      return (<Redirect to="/users/login" />);
    }
    return (
      <div id="add-admin-container">
        {/* Top navigation on small screen */}
        <UserTopNav
          name={this.props.user.name}
          title="Add an admin"
          isAdmin={this.props.isAdmin}
          isSuperAdmin={this.props.isSuperAdmin}
          dispatch={this.props.dispatch}
        />
        <div className="container-fluid">
          <div className="row">
            {/*  Side navigation on large screen */}
            <UserSideNav
              name={this.props.user.name}
              isAdmin={this.props.isAdmin}
              isSuperAdmin={this.props.isSuperAdmin}
              dispatch={this.props.dispatch}
            />
            {/* Main content */}
            <div className="col-lg-10 offset-md-2 mt-lg-0" id="main-content">
              {/* Content Header(navigation) on large screen */}
              <Header text="Add an admin" />
              <div className="d-flex flex-column align-items-center mt-5">
                <LoadingIcon start={this.props.addingAdminStarted} size={3} />
                <div className="card align-items-center text-center w-75">
                  <div className="card-body">
                    <BigAlert message={this.props.addingAdminError} />
                    {
                      this.props.addingAdminResolved ?
                        <BigAlert
                          message={`${this.state.email} is now an admin`}
                          type="success"
                        /> : null
                    }
                    <div className="input-group px-3">
                      <span className="input-group-addon">
                        <i className="fa fa-user" />
                      </span>
                      <input
                        id="input-email"
                        type="text"
                        className="form-control"
                        placeholder="The user's email"
                        name="email"
                        onChange={this.getInput}
                      />
                    </div>
                    <SmallAlert message={this.state.inputErrors.emailError} />
                    <ul className="text-left mt-2 text-muted">
                      <li>The user must have signed up already</li>
                      <li>If successful, the user would have access to:
                          <ol type="number">
                          <li>Creating a Center</li>
                          <li>Editing a Center</li>
                          <li>Viewing transactions</li>
                          <li>Making decisions on transactions</li>
                        </ol>
                      </li>
                    </ul>
                    <button
                      id="add-btn"
                      className="btn btn-primary pointer-button"
                      disabled={this.props.addingAdminStarted}
                      onClick={this.add}
                    >Add Admin
                      </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <footer className="d-block d-sm-none mt-5">
          <div className="container text-white text-center py-5">
            <h1>Ievents</h1>
            <p>Copyright &copy; 2017</p>
          </div>
        </footer>
      </div>
    );
  }
}

export default AddAdmin;
