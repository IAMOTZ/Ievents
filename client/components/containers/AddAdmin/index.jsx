import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { validateAddAdminInputs } from '../../../helpers/inputValidators';
import { addAdmin } from '../../../actions/userActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import View from './View';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    addingAdminStarted: store.addAdminReducer.addingAdminStarted,
    addingAdminResolved: store.addAdminReducer.addingAdminResolved,
    addingAdminError: store.addAdminReducer.addingAdminError,
  };
})
class AddAdmin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      inputErrors: {
        emailError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_ADMIN));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    if (this.props.addingAdminResolved || this.props.addingAdminError) {
      this.refresh();
    }
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      emailError: null,
    };
    this.setState(state);
  }

  /**
   * Refresh the page by stopping any asynchronous process that might be going on.
   */
  refresh() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_ADMIN));
  }

  /**
   * Dispatches the action to add the admin.
   */
  add = () => {
    const { email } = this.state;
    const inputErrors = validateAddAdminInputs({ email });
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(addAdmin(this.state.email, this.props.userToken));
    }
  }

  render() {
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        addingAdminStarted={this.props.addingAdminStarted}
        addingAdminError={this.props.addingAdminError}
        addingAdminResolved={this.props.addingAdminResolved}
        newAdmin={this.state.email}
        getInput={this.getInput}
        inputErrors={this.state.inputErrors}
        add={this.add}
      />
    );
  }
}

AddAdmin.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  addingAdminStarted: false,
  addingAdminResolved: false,
  addingAdminError: '',
  dispatch: () => {},
};

AddAdmin.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  addingAdminStarted: PropTypes.bool,
  addingAdminResolved: PropTypes.bool,
  addingAdminError: PropTypes.string,
  dispatch: PropTypes.func,
};

export default AddAdmin;
