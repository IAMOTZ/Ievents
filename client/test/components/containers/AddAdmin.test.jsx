import React from 'react';
import AddAdmin from '../../../components/containers/AddAdmin';
import AddAdminView from '../../../components/containers/AddAdmin/View';


describe('<AddAdmin />', () => {
  const props = {
    userName: 'tester',
    userToken: 'justASampleToken',
    isAdmin: false,
    isSuperAdmin: false,
    addingAdminStarted: false,
    addingAdminResolved: false,
    addingAdminError: null,
    dispatch: jest.fn()
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AddAdmin {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<AddAdmin {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('GET_INPUT', () => {
      it('should update the state with user input', () => {
        const userInput = 'tester@gmail.com';
        wrapperInstance.getInput({ target: { name: 'email', value: userInput } });
        expect(wrapper.state('email')).toEqual(userInput);
      });
      it('should dispatch an action if adding admin already resolved', () => {
        wrapper.setProps(alterProps({ addingAdminResolved: true }));
        wrapperInstance.getInput({});
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('CLEAR_INPUT_ERRORS', () => {
      it('should clear input errors in the state', () => {
        const emailError = 'Email is required';
        wrapper.setState({ inputErrors: { emailError } });
        wrapperInstance.clearInputErrors();
        expect(wrapper.state('inputErrors').emailError).toBeNull();
      });
    });

    describe('ADD', () => {
      it('should set inputErros if validation fails', () => {
        const wrongEmail = 'testergmail.com';
        wrapper.setState({ email: wrongEmail });
        wrapperInstance.add();
        expect(typeof wrapper.state('inputErrors').emailError).toEqual('string');
      });
      it('should clear inputErrors and dispatch action if validation pass', () => {
        const correctEmail = 'tester@gmail.com';
        const previousError = 'Email is required';
        wrapper.setState({
          email: correctEmail, inputErrors: { emailError: previousError }
        });
        wrapperInstance.add();
        expect(wrapper.state('inputErrors').emailError).toBeNull();
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('REFRESH', () => {
      it('should dispatch an action', () => {
        wrapperInstance.refresh();
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('COMOPONENT_WILL_UNMOUNT', () => {
      it('should dispatch an action when the component wants to unmount', () => {
        wrapper.unmount();
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
  });
});

describe('<AddAdminView />', () => {
  const props = {
    add: jest.fn(),
    addingAdminError: null,
    addingAdminStarted: false,
    addingAdminResolved: false,
    dispatch: jest.fn(),
    getInput: jest.fn(),
    isAdmin: false,
    isSuperAdmin: false,
    newAdmin: 'newAdmin@gmail.com',
    userName: 'tester',
    userToken: 'justASampleToken',
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AddAdminView {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render an alert when adding admin resolves', () => {
      wrapper.setProps(alterProps({ addingAdminResolved: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
