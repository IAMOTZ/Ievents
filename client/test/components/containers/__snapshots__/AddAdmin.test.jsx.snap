// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddAdmin /> Rendering: should render correctly 1`] = `
<View
  add={[Function]}
  addingAdminError={null}
  addingAdminResolved={false}
  addingAdminStarted={false}
  dispatch={[MockFunction]}
  getInput={[Function]}
  inputErrors={
    Object {
      "emailError": null,
    }
  }
  isAdmin={false}
  isSuperAdmin={false}
  newAdmin={null}
  userName="tester"
/>
`;

exports[`<AddAdminView /> Rendering should render an alert when adding admin resolves 1`] = `
<div
  id="add-admin-container"
>
  <SideNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <AuthTopNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
        title="Add an admin"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="Add an admin"
        />
        <div
          className="page-content"
        >
          <div
            className="d-flex flex-column align-items-center mt-5"
          >
            <LoadingIcon
              size={3}
              start={false}
            />
            <div
              className="card align-items-center text-center w-75"
            >
              <div
                className="card-body"
              >
                <BigAlert
                  message={null}
                />
                <BigAlert
                  message="newAdmin@gmail.com is now an admin"
                  type="success"
                />
                <div
                  className="input-group px-3"
                >
                  <span
                    className="input-group-addon"
                  >
                    <i
                      className="fa fa-envelope"
                    />
                  </span>
                  <input
                    className="form-control"
                    id="input-email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="The user's email"
                    type="text"
                  />
                </div>
                <SmallAlert />
                <ul
                  className="text-left mt-2"
                >
                  <li>
                    The user must have signed up already
                  </li>
                  <li>
                    If successful, the user would have access to:
                    <ol
                      type="number"
                    >
                      <li>
                        Creating a Center
                      </li>
                      <li>
                        Editing a Center
                      </li>
                      <li>
                        Viewing transactions
                      </li>
                      <li>
                        Making decisions on transactions
                      </li>
                    </ol>
                  </li>
                </ul>
                <button
                  className="btn ie-blue-button"
                  disabled={false}
                  id="add-btn"
                  onClick={[MockFunction]}
                >
                  Add Admin
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;

exports[`<AddAdminView /> Rendering should render correctly 1`] = `
<div
  id="add-admin-container"
>
  <SideNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <AuthTopNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
        title="Add an admin"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="Add an admin"
        />
        <div
          className="page-content"
        >
          <div
            className="d-flex flex-column align-items-center mt-5"
          >
            <LoadingIcon
              size={3}
              start={false}
            />
            <div
              className="card align-items-center text-center w-75"
            >
              <div
                className="card-body"
              >
                <BigAlert
                  message={null}
                />
                <div
                  className="input-group px-3"
                >
                  <span
                    className="input-group-addon"
                  >
                    <i
                      className="fa fa-envelope"
                    />
                  </span>
                  <input
                    className="form-control"
                    id="input-email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="The user's email"
                    type="text"
                  />
                </div>
                <SmallAlert />
                <ul
                  className="text-left mt-2"
                >
                  <li>
                    The user must have signed up already
                  </li>
                  <li>
                    If successful, the user would have access to:
                    <ol
                      type="number"
                    >
                      <li>
                        Creating a Center
                      </li>
                      <li>
                        Editing a Center
                      </li>
                      <li>
                        Viewing transactions
                      </li>
                      <li>
                        Making decisions on transactions
                      </li>
                    </ol>
                  </li>
                </ul>
                <button
                  className="btn ie-blue-button"
                  disabled={false}
                  id="add-btn"
                  onClick={[MockFunction]}
                >
                  Add Admin
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;
